src/js/utils/JobUtil.js

```diff
@@ -78,18 +78,27 @@ const JobUtil = {
       Object.assign(spec.run, { docker });
     }
 
+    // default
+    var scheduleId = "default";
+    var schedulePolicy = "ALLOW";
+    // preserve id and concurrencyPolicy
+    if (typeof spec.schedules != "undefined" && spec.schedules.length > 0) {
+      scheduleId = spec.schedules[0].id;
+      schedulePolicy = spec.schedules[0].concurrencyPolicy;
```

Comment:
I just wanna check the assumption here that the schedule[0] is present, then we will always have Id and Policy. Otherwise, this might cause a bug where we have an empty schedule policy and the default is ignored.

Reply:
if `schedules` is defined (it is an array) and it has at least 1 element... that element will be a schedule.   that schedule will always have an id (because it is required) and it should have a concurrencyPolicy.   I'll check to see if it is possibly for this to be undefined or empty.

Reply:
yeah... good catch... after some thinking on this... I know it is possible for this to be missing... I'll update.

Reply:
checked and verified... metronome automatically provides the ALLOW concurrencyPolicy if a policy isn't specified.  this value will always be defined here.

Reply:
although this tests out fine on metronome... it is possible (as I've tested) that the UI user switches to the JSON edit, removes it, switches to UI edit, then switches back to JSON... this will result in an undefined element :( .   Seems unlikely so you know it will happen.

Reply:
haha... will oddly the UI code already handles this in an ok way... the event that follows this code under these circumstances is `createJobSpecFromJob` which defaults the concurrencyPolicy if not provided.
although it is possible for this value to be undefined (in the context of the UI alone)...  it is handled well within the UI.   If sent to Metronome it is defaulted there.   We could add defensive code, but it would convolute the code in a way that isn't necessary... open to other opinions / preferences.